### ISSUE TITLE
Can't update flynn-host: too many open files

### ISSUE THREAD

**Issue by Alir3z4** [flynn/flynn#3714]
Hello, I've tried to update the flynn-host to the newest version.
I've run the update, but it fails:
Error: **Server error: error downloading layer: pipe2: too many open files:**

Here's the update command result:
```
ubuntu@ip-10-0-3-163:~$ sudo flynn-host update
INFO[11-19|06:36:51] initializing TUF client 
INFO[11-19|06:36:51] updating TUF data 
INFO[11-19|06:36:52] getting configured release channel 
INFO[11-19|06:36:52] determining latest version of stable release channel 
INFO[11-19|06:36:52] latest stable version is v20161114.0p1 
INFO[11-19|06:36:52] downloading v20161114.0p1 flynn-host binary 
INFO[11-19|06:36:52] executing latest flynn-host binary 
INFO[11-19|06:36:52] initializing TUF client 
INFO[11-19|06:36:52] reading TUF database 
INFO[11-19|06:36:52] getting host list 
INFO[11-19|06:36:52] updating 1 hosts 
INFO[11-19|06:36:52] checking host version compatibility 
INFO[11-19|06:36:52] pulling latest images on all hosts 
INFO[11-19|06:36:52] pulling images                           host=ip1003163
INFO[11-19|06:36:53] pulling postgres image                   host=ip1003163
INFO[11-19|06:36:53] pulling postgres layer 7e815919b7a6c176ed286f20a06dfb038167be2a66d8faa0649e68c80df0d97c (62.3 MiB) host=ip1003163
INFO[11-19|06:36:53] pulling postgres layer 0411781f64ff4cb35400ceca89e9f67d2df2f017340bb25d99f555f1e887ca44 (112 MiB) host=ip1003163
EROR[11-19|06:36:55] error pulling images                     host=ip1003163 err="Server error: error downloading layer: pipe2: too many open files: \"/sbin/zfs zfs create -p -V 117432320 -o volblocksize=8192 flynn-default/squashfs/0411781f64ff4cb35400ceca89e9f67d2df2f017340bb25d99f555f1e887ca44\" => "
06:36:55.628310 host.go:157: Server error: error downloading layer: pipe2: too many open files: "/sbin/zfs zfs create -p -V 117432320 -o volblocksize=8192 flynn-default/squashfs/0411781f64ff4cb35400ceca89e9f67d2df2f017340bb25d99f555f1e887ca44" => 
```

Current version:
```
ubuntu@ip-10-0-3-163:~$ sudo flynn-host version
v20161114.0
```

Collected debug logs: https://gist.github.com/c472b6dbe5f6bbfce45213e31579a8ee

**Comment by Alir3z4** [comment]
I've restarted the the node (the cluster is an one node cluster) and tried to update again, this time a timeout:

Update command result:
```
root@ip-10-0-3-163:/home/ubuntu# flynn-host update
INFO[11-21|04:17:30] initializing TUF client 
INFO[11-21|04:17:30] updating TUF data 
INFO[11-21|04:17:35] getting configured release channel 
INFO[11-21|04:17:35] determining latest version of stable release channel 
INFO[11-21|04:17:35] latest stable version is v20161114.0p1 
INFO[11-21|04:17:35] downloading v20161114.0p1 flynn-host binary 
INFO[11-21|04:17:35] executing latest flynn-host binary 
INFO[11-21|04:17:35] initializing TUF client 
INFO[11-21|04:17:35] reading TUF database 
INFO[11-21|04:17:35] getting host list 
INFO[11-21|04:17:35] updating 1 hosts 
INFO[11-21|04:17:35] checking host version compatibility 
INFO[11-21|04:17:35] pulling latest images on all hosts 
INFO[11-21|04:17:35] pulling images                           host=ip1003163
INFO[11-21|04:17:37] pulling flannel image                    host=ip1003163
INFO[11-21|04:17:37] pulling flannel layer 0e18841a52b6f3c1ca4e6538de38d57d50ef572d9a65844060842a6268a47bd7 (1.973 MiB) host=ip1003163
INFO[11-21|04:17:37] pulling flannel layer 9034bacb1d298a48d343d54685b25b98c549ad4a06597e9adbb3924d15ff5a42 (5.137 MiB) host=ip1003163
INFO[11-21|04:17:37] pulling blobstore image                  host=ip1003163
INFO[11-21|04:17:37] pulling blobstore layer 7e815919b7a6c176ed286f20a06dfb038167be2a66d8faa0649e68c80df0d97c (62.3 MiB) host=ip1003163
INFO[11-21|04:17:37] pulling blobstore layer f773c3fa1c9d0bf4fb5392c78cecfdaa7f4ad4645345e0f7ea86568b6d80c8bc (29.78 MiB) host=ip1003163
INFO[11-21|04:17:38] pulling slugrunner image                 host=ip1003163
INFO[11-21|04:17:38] pulling slugrunner layer 7e815919b7a6c176ed286f20a06dfb038167be2a66d8faa0649e68c80df0d97c (62.3 MiB) host=ip1003163
INFO[11-21|04:17:38] pulling slugrunner layer 4c1cc7b8a1a83a1ecc0cde8ab6ec0e41b8e718d6f964486244952d99d329f201 (433.8 MiB) host=ip1003163
INFO[11-21|04:17:38] pulling slugrunner layer 18962e0ce642ebcdb47b50e63c94d439141ce86a77d9d923e90ea0fd3cfc13e5 (4 KiB) host=ip1003163
INFO[11-21|04:17:38] pulling status image                     host=ip1003163
INFO[11-21|04:17:38] pulling status layer 0e18841a52b6f3c1ca4e6538de38d57d50ef572d9a65844060842a6268a47bd7 (1.973 MiB) host=ip1003163
INFO[11-21|04:17:38] pulling status layer 8cc5379643ec92bc75e8e2956557d752faa1f8186421f016ea4e8a4ee6c66ff1 (2.496 MiB) host=ip1003163
INFO[11-21|04:17:39] pulling mongodb image                    host=ip1003163
INFO[11-21|04:17:39] pulling mongodb layer 53ac780c79b5d20722c0839bceba2ed4109d9004fe025ac7dee13d9df84b3f95 (32.97 MiB) host=ip1003163
INFO[11-21|04:17:39] pulling mongodb layer 380b5e8059a21484ab6661b2e51ce4ff977407972a376da72c1362af3051bd8c (85.51 MiB) host=ip1003163
INFO[11-21|04:17:41] pulling discoverd image                  host=ip1003163
INFO[11-21|04:17:41] pulling discoverd layer 0e18841a52b6f3c1ca4e6538de38d57d50ef572d9a65844060842a6268a47bd7 (1.973 MiB) host=ip1003163
INFO[11-21|04:17:41] pulling discoverd layer ed64b5837d3fbdcad1c3443121c137db44a0c358fd6504867a087f0852776326 (3.742 MiB) host=ip1003163
INFO[11-21|04:17:41] pulling gitreceive image                 host=ip1003163
INFO[11-21|04:17:41] pulling gitreceive layer 7e815919b7a6c176ed286f20a06dfb038167be2a66d8faa0649e68c80df0d97c (62.3 MiB) host=ip1003163
INFO[11-21|04:17:41] pulling gitreceive layer 2eddb9ce6a60e74c1afc3275aa5e4b4a55addf0bebd73c4090dc308bc02dab60 (41.09 MiB) host=ip1003163
INFO[11-21|04:17:43] pulling dashboard image                  host=ip1003163
INFO[11-21|04:17:43] pulling dashboard layer 0e18841a52b6f3c1ca4e6538de38d57d50ef572d9a65844060842a6268a47bd7 (1.973 MiB) host=ip1003163
INFO[11-21|04:17:43] pulling dashboard layer 5a4d4095ae2ee79d7766b63a4501cebe577a26ec97b18e129a9f853264ed110f (6.758 MiB) host=ip1003163
INFO[11-21|04:17:43] pulling logaggregator image              host=ip1003163
INFO[11-21|04:17:43] pulling logaggregator layer 0e18841a52b6f3c1ca4e6538de38d57d50ef572d9a65844060842a6268a47bd7 (1.973 MiB) host=ip1003163
INFO[11-21|04:17:43] pulling logaggregator layer 0ec626fa5b44062fc3cb5dfdf0fb21ab10fb027542e23ccdb6dbc26b348ba0d4 (2.699 MiB) host=ip1003163
INFO[11-21|04:17:44] pulling docker-receive image             host=ip1003163
INFO[11-21|04:17:44] pulling docker-receive layer 7e815919b7a6c176ed286f20a06dfb038167be2a66d8faa0649e68c80df0d97c (62.3 MiB) host=ip1003163
INFO[11-21|04:17:44] pulling docker-receive layer 96b1d8adcd0082f2a5c4e4e3406f08f61233c5a5bbcf7a10f811a8eee1ef45cc (10.08 MiB) host=ip1003163
INFO[11-21|04:17:44] pulling controller image                 host=ip1003163
INFO[11-21|04:17:44] pulling controller layer 0e18841a52b6f3c1ca4e6538de38d57d50ef572d9a65844060842a6268a47bd7 (1.973 MiB) host=ip1003163
INFO[11-21|04:17:44] pulling controller layer 5a4aa5d03958b9dec616227716034e0efbb3f8137db24168f79d4dbf9ab17b74 (9.047 MiB) host=ip1003163
INFO[11-21|04:17:45] pulling router image                     host=ip1003163
INFO[11-21|04:17:45] pulling router layer 0e18841a52b6f3c1ca4e6538de38d57d50ef572d9a65844060842a6268a47bd7 (1.973 MiB) host=ip1003163
INFO[11-21|04:17:45] pulling router layer 464270747007b66e3bcef0ecf3b8b81bc90b30e821f39b27393527405bf19d45 (3.016 MiB) host=ip1003163
INFO[11-21|04:17:46] pulling taffy image                      host=ip1003163
INFO[11-21|04:17:46] pulling taffy layer 7e815919b7a6c176ed286f20a06dfb038167be2a66d8faa0649e68c80df0d97c (62.3 MiB) host=ip1003163
INFO[11-21|04:17:46] pulling taffy layer 20ff678860bd75ba738ef6b5331ef7f94547c6fdf75f37e291e90d7f2a26947c (39.28 MiB) host=ip1003163
INFO[11-21|04:17:47] pulling redis image                      host=ip1003163
INFO[11-21|04:17:47] pulling redis layer 7e815919b7a6c176ed286f20a06dfb038167be2a66d8faa0649e68c80df0d97c (62.3 MiB) host=ip1003163
INFO[11-21|04:17:47] pulling redis layer b321fb0f7e044a3223bd1d378514180ab612739fba9a92f898f8825ea0347491 (78.31 MiB) host=ip1003163
INFO[11-21|04:17:49] pulling mariadb image                    host=ip1003163
INFO[11-21|04:17:49] pulling mariadb layer 53ac780c79b5d20722c0839bceba2ed4109d9004fe025ac7dee13d9df84b3f95 (32.97 MiB) host=ip1003163
INFO[11-21|04:17:49] pulling mariadb layer 76701464cfe5b0eb6abce8412034c4029d347fe9b0bbb764189ca8dc140d6d9c (108.2 MiB) host=ip1003163
INFO[11-21|04:17:52] pulling postgres image                   host=ip1003163
INFO[11-21|04:17:52] pulling postgres layer 7e815919b7a6c176ed286f20a06dfb038167be2a66d8faa0649e68c80df0d97c (62.3 MiB) host=ip1003163
INFO[11-21|04:17:52] pulling postgres layer 0411781f64ff4cb35400ceca89e9f67d2df2f017340bb25d99f555f1e887ca44 (112 MiB) host=ip1003163
INFO[11-21|04:17:55] pulling slugbuilder image                host=ip1003163
INFO[11-21|04:17:55] pulling slugbuilder layer 7e815919b7a6c176ed286f20a06dfb038167be2a66d8faa0649e68c80df0d97c (62.3 MiB) host=ip1003163
INFO[11-21|04:17:55] pulling slugbuilder layer 4c1cc7b8a1a83a1ecc0cde8ab6ec0e41b8e718d6f964486244952d99d329f201 (433.8 MiB) host=ip1003163
INFO[11-21|04:17:55] pulling slugbuilder layer e35e6873750f04353930a275fc17a395fb04e0d354bdf2868514ddbc05721a13 (17.61 MiB) host=ip1003163
INFO[11-21|04:17:56] pulling updater image                    host=ip1003163
INFO[11-21|04:17:56] pulling updater layer 0e18841a52b6f3c1ca4e6538de38d57d50ef572d9a65844060842a6268a47bd7 (1.973 MiB) host=ip1003163
INFO[11-21|04:17:56] pulling updater layer 1e59dc337d9207c2d882a88920485454df0b6c4b8dbf421490bb3e6dc0f544d2 (2.445 MiB) host=ip1003163
INFO[11-21|04:17:57] pulling latest binaries and config on all hosts 
INFO[11-21|04:17:57] pulling binaries and config              host=ip1003163
INFO[11-21|04:18:00] binaries and config pulled successfully  host=ip1003163
INFO[11-21|04:18:00] validating binaries 
INFO[11-21|04:18:00] updating flynn-host daemon on all hosts 
INFO[11-21|04:18:01] flynn-host updated successfully          host=ip1003163
EROR[11-21|04:18:01] error getting cluster status             err="Get http://status-web.discoverd: dial tcp: lookup status-web.discoverd on 100.100.20.1:53: no such host"
```

Debug information: https://gist.github.com/ef436947f8a02e904970bdba197d5976

**Comment by titanous** [comment]
This was fixed by #3771.
